Sources/Snowplow/Tracker/SessionState.swift

```diff
@@ -0,0 +1,107 @@
+//
+//  SPSessionState.swift
+//  Snowplow
+//
+//  Copyright (c) 2013-2022 Snowplow Analytics Ltd. All rights reserved.
+//
+//  This program is licensed to you under the Apache License Version 2.0,
+//  and you may not use this file except in compliance with the Apache License
+//  Version 2.0. You may obtain a copy of the Apache License Version 2.0 at
+//  http://www.apache.org/licenses/LICENSE-2.0.
+//
+//  Unless required by applicable law or agreed to in writing,
+//  software distributed under the Apache License Version 2.0 is distributed on
+//  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+//  express or implied. See the Apache License Version 2.0 for the specific
+//  language governing permissions and limitations there under.
+//
+//  Authors: Alex Benini
+//  License: Apache License Version 2.0
+//
+
+import Foundation
+
+@objc(SPSessionState)
+public class SessionState: NSObject, State {
+    @objc
+    public private(set) var firstEventId: String?
+    @objc
+    public private(set) var firstEventTimestamp: String?
```

Comment:
question: why is `firstEventTimestamp` optional? I have it as just `String` right now in Kotlin

Reply:
I changed it to be optional here because it is optional in the `client_session` schema so wanted to match that. But in practice it will never be null because the first event timestamp is always known (also the first event ID).